aioamqp/channel.py

```diff
@@ -480,7 +484,7 @@ def basic_publish(self, payload, exchange_name, routing_key, properties=None, ma
         if isinstance(payload, str):
             warnings.warn("Str payload support will be removed in next release", DeprecationWarning)
             payload = payload.encode()
-
+        payload_len = len(payload)
```

Comment:
not sure it's an optimization here

Reply:
Every function call takes a little time. But in this case it's like a drop in the sea. I will revert this changes.